src/app/domain/robot.type.ts

```diff
@@ -24,13 +24,13 @@ export class RobotType {
         public protocol: typeof BaseProtocol,
         public name: string,
         public svgname: string,
-        public micrcontroller: string,
         public fqbn: string,
         public ext: string,
         public core: string,
         public libs: string[],
         public isWired: boolean = true,
-        features?: Partial<Features>
+        features?: Partial<Features>,
+        public micrcontroller?: string
```

Comment:
I would still prefer the solution with metadata, since it would reduce the insane number of current arguments

Reply:
Ok for now, will be improved later